src/interface/viewport.c

```diff
@@ -1398,28 +1411,34 @@ void viewport_track_paint_setup(uint8 direction, int height, rct_map_element *ma
 			RCT2_GLOBAL(0x00F441A4, uint32) = 0x21600000;
 		}
 		if (mapElement->flags & MAP_ELEMENT_FLAG_GHOST) {
-			uint32 meh = RCT2_ADDRESS(0x00993CC4, uint32)[RCT2_GLOBAL(0x009AACBF, uint8)];
+			uint32 ghost_id = RCT2_ADDRESS(0x00993CC4, uint32)[RCT2_GLOBAL(0x009AACBF, uint8)];
 			RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = 0;
-			RCT2_GLOBAL(0x00F44198, uint32) = meh;
-			RCT2_GLOBAL(0x00F4419C, uint32) = meh;
-			RCT2_GLOBAL(0x00F441A0, uint32) = meh;
-			RCT2_GLOBAL(0x00F441A4, uint32) = meh;
+			RCT2_GLOBAL(0x00F44198, uint32) = ghost_id;
+			RCT2_GLOBAL(0x00F4419C, uint32) = ghost_id;
+			RCT2_GLOBAL(0x00F441A0, uint32) = ghost_id;
+			RCT2_GLOBAL(0x00F441A4, uint32) = ghost_id;
 		}
 
-		uint32 **trackTypeList = (uint32**)RideTypeTrackPaintFunctions[ride->type];
-		uint32 *trackDirectionList = trackTypeList[trackType];
-
-		// Have to call from this point as it pushes esi and expects callee to pop it
-		RCT2_CALLPROC_X(
-			0x006C4934,
-			ride->type,
-			(int)trackDirectionList,
-			direction,
-			height,
-			(int)mapElement,
-			rideIndex * sizeof(rct_ride),
-			trackSequence
-		);
+		TRACK_PAINT_FUNCTION ***trackTypeList = (TRACK_PAINT_FUNCTION***)RideTypeTrackPaintFunctionsOld[ride->type];
+		if (trackTypeList == NULL) {
+			trackTypeList = (TRACK_PAINT_FUNCTION***)RideTypeTrackPaintFunctions[ride->type];
+			trackTypeList[trackType][direction](rideIndex, trackSequence, direction, height, mapElement);
```

Comment:
If we fetch this and then check for NULL here, we can eliminate using the dummy function.